src/github.rs

```diff
@@ -0,0 +1,272 @@
+use crate::install::AssetType;
+use color_eyre::{eyre::eyre, Result};
+use reqwest::Client;
+use serde_json::Value;
+
+pub struct GithubReleaseRepository {
+    pub base_url: String,
+    pub organisation_name: String,
+    pub repository_name: String,
+}
+
+impl GithubReleaseRepository {
+    pub fn new(base_url: &str, organisation_name: &str, repository_name: &str) -> Self {
+        Self {
+            base_url: base_url.to_string(),
+            organisation_name: organisation_name.to_string(),
+            repository_name: repository_name.to_string(),
+        }
+    }
+
+    /// Retrieves the asset name of the latest release for the specified type and platform.
+    ///
+    /// # Arguments
+    ///
+    /// * `asset_type` - Either the client or node.
+    /// * `platform` - The target triple platform of the binary to be installed.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing the name of the asset and the version number for the latest release.
+    ///
+    /// # Errors
+    ///
+    /// Returns an `eyre::Report` if an error occurs during the retrieval process. Possible error
+    /// conditions include network errors, JSON parsing errors, and missing or invalid data in the
+    /// response body.
+    pub async fn get_latest_asset_name(
+        &self,
+        asset_type: AssetType,
+        platform: &str,
+    ) -> Result<(String, String)> {
+        let client = Client::new();
+        let response = client
+            .get(format!(
+                "{}/repos/{}/{}/releases/latest",
+                self.base_url, self.organisation_name, self.repository_name
+            ))
+            .header("User-Agent", "safeup")
+            .send()
+            .await?;
+        let body = response.text().await?;
+        let json: Value = serde_json::from_str(&body)?;
+        let tag_name = json["tag_name"]
+            .as_str()
+            .ok_or_else(|| eyre!("Response body does not contain 'tag_name' value"))?;
+        let version = self.get_version_from_tag_name(&asset_type, tag_name)?;
+        let asset_name = match asset_type {
+            AssetType::Client => format!("sn_cli-{version}-{platform}.tar.gz"),
+            AssetType::Node => {
+                format!("sn_node-{version}-{platform}.tar.gz")
+            }
+        };
+        if self.release_has_asset(&json, &asset_name)? {
+            return Ok((asset_name, version));
+        }
+
+        let msg = match asset_type {
+            AssetType::Client => format!("Release has no client asset for platform {platform}"),
+            AssetType::Node => format!("Release has no node asset for platform {platform}"),
+        };
+        Err(eyre!(msg))
```

Comment:
Also, may be just my preference, but I like to keep the return statement separated by an empty line as well, as to give that extra enhancement for distinguishing it in quick glances / perusing swathes of code.